backend/src/main/scala/cromwell/backend/BackendWorkflowInitializationActor.scala

```diff
@@ -2,58 +2,127 @@ package cromwell.backend
+import scalaz.{Validation, Failure, NonEmptyList}
 
 object BackendWorkflowInitializationActor {
 
   // Commands
   sealed trait BackendWorkflowInitializationActorCommand extends BackendWorkflowLifecycleActorCommand
-
   case object Initialize extends BackendWorkflowInitializationActorCommand
   final case class Abort(jobKey: BackendJobDescriptorKey) extends BackendWorkflowInitializationActorCommand
 
   // Responses
   sealed trait BackendWorkflowInitializationActorResponse extends BackendWorkflowLifecycleActorResponse
-
   sealed trait InitializationResponse extends BackendWorkflowInitializationActorResponse
   case object InitializationSuccess extends InitializationResponse
   case class InitializationFailed(reason: Throwable) extends Exception with InitializationResponse
+
 }
 
 /**
   * Workflow-level actor for executing, recovering and aborting jobs.
   */
 trait BackendWorkflowInitializationActor extends BackendWorkflowLifecycleActor with ActorLogging {
 
-  final def receive: Receive = LoggingReceive {
-    case Initialize    => performActionThenRespond(initSequence, onFailure = InitializationFailed)
-    case AbortWorkflow => performActionThenRespond(abortInitialization, onFailure = BackendWorkflowAbortFailedResponse)
+  def receive: Receive = LoggingReceive {
+    case Initialize => performActionThenRespond(initSequence(), onFailure = InitializationFailed)
+    case AbortWorkflow => performActionThenRespond(abortInitialization(), onFailure = BackendWorkflowAbortFailedResponse)
   }
 
   /**
     * Our predefined sequence to run during preStart
     */
-  final def initSequence = for {
-    _ <- validate
-    _ <- beforeAll
+  final def initSequence() = for {
+    _ <- validate()
+    _ <- beforeAll()
   } yield InitializationSuccess
 
   /**
     * Abort all initializations.
     */
-  def abortInitialization: Future[WorkflowAbortResponse]
+  def abortInitialization(): Future[WorkflowAbortResponse]
+
+  /**
+    * A call which happens before anything else runs
+    */
+  def beforeAll(): Future[Unit]
+
+  /**
+    * Validates runtime attributes for one specific call.
+    *
+    * @param runtimeAttributes Runtime Attributes with already evaluated values.
+    * @return If all entries from runtime attributes section are valid Success otherwise
+    *         Failure with the aggregation of errors.
+    */
+  def validateRuntimeAttributes(runtimeAttributes: EvaluatedRuntimeAttributes): Future[scalaz.Validation[NonEmptyList[String], Unit]]
 
   /**
     * Validate that this WorkflowBackendActor can run all of the calls that it's been assigned
     */
-  def validate: Future[Unit]
+  protected def validate(): Future[Unit] = {
+    def eval(call: Call) = validateRuntimeAttributes(evaluateRuntimeAttributesWdlExpressions(call))
+
+    val validationResult: Future[Seq[(Call, Validation[NonEmptyList[String], Unit])]] =
+      Future.sequence(calls map { call => (call, eval(call)) } map { entry => entry._2.map(valRes => (entry._1, valRes)) })
```

Comment:
The middle map seems superfluous:
Would just `Future.sequence(calls map { call => eval(call).map(valRes => (call, valRes)) })` work?